cpp/ps0007/150daysprogramming.cpp

```diff
@@ -26,158 +10,64 @@ float area(char shape, float dimensions[]);
-        break;
-    case 's':
-        cout << " enter the square side length: ";
-        cin >> input1;
-
-
-        if (input1 > 0) {
-            dimensions[0] = input1;
-
-          
-        }
-        return  dimensions[0] * dimensions[0];
-        break;
-    case 't':
-        cout << " enter the triangle base length: ";
-        cin >> input1;
-
-        cout << " enter the triangle height length: ";
-        cin >> input2;
-
-
-        if (input1 > 0 && input2 > 0) {
-            dimensions[0] = input1;
-            dimensions[1] = input2;
-
-           
-        }
-        return  dimensions[0] * dimensions[1]/2;
-        break;
-    default:
-        cout << "wrong charcter ";
-    }
-
-
-
-
-
-
-
-
-}
-
-void special_counter(int x) {
-    if (x > 0) {
-        for (int i = 0; i <= x; i++) {
-            cout << i << "\t";
-        }
-    }
-    else
-    {
-        for (int i = 0; i >= x; i--) {
-            cout << i << "\t";
-        }
-
-    }
-    
-}
-
-void special_counter(int end , int start) {
-    if (end > start) {
-        for (int i = start; i <= end; i++) {
-            cout << i << "\t";
-        }
-    }
-    else {
-        for (int i = start; i >= end; i--) {
-            cout << i << "\t";
-        }
-       // cout << "start must be less than end";
-    }
-   
-
-}
-
-void special_counter(int end , int start , int step) {
-    if (end > start) {
-        for (int i = start; i <= end; i += step) {
-            cout << i << "\t";
-        }
-    }
-    else {
-        for (int i = start; i >= end; i+=step) {
-            cout << i << "\t";
-        }
-      //  cout << "start must be less than end";
-    }
-
+        case 'c':
+            cout << " enter the circle radius: ";
+            cin >> dimensions[0];
+            if (dimensions[0] > 0) {
+                return PI * dimensions[0] * dimensions[0];
+            }
+            break;
+        
+        case 'r':
+            cout << " enter the rectangle length: ";
+            cin >> dimensions[0];
+            cout << " enter the rectangle width: ";
+            cin >> dimensions[1];
+            if (dimensions[0] > 0 && dimensions[1] > 0) {
+                return  dimensions[0] * dimensions[1];
+            }
+            break;
+        
+        case 's':
+            cout << " enter the square side length: ";
+            cin >> dimensions[0];
+            if (dimensions[0] > 0) {
+                return  dimensions[0] * dimensions[0];
+            }
+            break;
+        
+        case 't':
+            cout << " enter the triangle base length: ";
+            cin >> dimensions[0];
+            cout << " enter the triangle height length: ";
+            cin >> dimensions[1];
+            if (dimensions[0] > 0 && dimensions[1] > 0) {
+                return  dimensions[0] * dimensions[1] / 2;
+            }
+            break;
+        
+        default:
+            cout << "wrong charcter ";
+    } // dimensions must have positive value

```

Comment:
- [x] used suitable size of dimensions arrays with the shape
- [x] used `cout::std` in a right way to print out the area